lib/structures/GuildScheduledEvent.js

```diff
@@ -0,0 +1,154 @@
+* @prop {Guild | Object} guild The guild which the event belongs to. Can be partial with only `id` if not cached
+* @prop {String} id The id of the guild event
+* @prop {String?} image The hash of the event's image, or null if no image
+* @prop {String?} imageURL The URL of the event's image, or null if no image
+* @prop {String} name The name of the event
+* @prop {Number} privacyLevel Event privacy level
+* @prop {Number} scheduledStartTime The time the event will start
+* @prop {Number?} scheduledEndTime The time the event will end, or null if the event does not have a scheduled time to end
+* @prop {Number} status The [status](https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-guild-scheduled-event-status) of the scheduled event
+* @prop {Number?} userCount The number of users subscribed to the event
+*/
+class GuildScheduledEvent extends Base {
+    constructor(data, client) {
+        super(data.id);
+
+        this._client = client;
+        if(data.creator !== undefined) {
+            this.creator = client.users.update(data.creator, this.client);
+        } else {
+            this.creator = null;
+        }
+        this.guild = client.guilds.get(data.guild_id) || {id: data.guild_id};
+        this.scheduledEndTime = null;
+        this.update(data);
+    }
+
+    update(data) {
+        if(data.channel_id !== undefined) {
+            if(data.channel_id !== null) {
+                if(this._client.guilds.get(data.guild_id)) {
+                    this.channel = this._client.guilds.get(data.guild_id).channels.get(data.channel_id) || {id: data.channel_id};
+                } else {
+                    this.channel = {id: data.channel_id};
+                }
+            } else {
+                this.channel = null;
+            }
+        }
+        if(data.name !== undefined) {
+            this.name = data.name;
+        }
+        if(data.description !== undefined) {
+            this.description = data.description;
+        }
+        if(data.scheduled_start_time !== undefined) {
+            this.scheduledStartTime = Date.parse(data.scheduled_start_time);
+        }
+        if(data.scheduled_end_time !== undefined) {
+            this.scheduledEndTime = Date.parse(data.scheduled_end_time);
+        }
+        if(data.privacy_level !== undefined) {
+            this.privacyLevel = data.privacy_level;
+        }
+        if(data.status !== undefined) {
+            this.status = data.status;
+        }
+        if(data.entity_type !== undefined) {
+            this.entityType = data.entity_type;
+        }
+        if(data.entity_id !== undefined) {
+            this.entityID = data.entity_id;
+        }
+        if(data.entity_metadata !== undefined) {
+            this.entityMetadata = {location: data.location};
```

Comment:
```suggestion
            this.entityMetadata = {
                location: data.entity_metadata.location
            };
```
 
Since `entity_metadata` currently only includes `location` (which is optional) which is the same as above, you could also just use:
```suggestion
            this.entityMetadata = data.entity_metadata;
```